knitr/planetary_motion/planetary_motion.rmd

```diff
@@ -0,0 +1,735 @@
+
+## Equations of motion
+
+A more natural treatment of the problem would be using polar coordinates, but for simplicity, let's use Cartesian coordinates.
+The motion of the planet is described by Newton's law of motion, which is a second-order differential equation.
+For convenience, we use Hamilton's formulation of the law, which is based on a system of two first-order differential equations,
+\begin{eqnarray*}
+  \frac{\mathrm d q}{\mathrm d t} & = & \frac{p}{m}, \\
+  \frac{\mathrm d q}{\mathrm d t} & = & - \frac{k}{r^3} (q - q_*),
+\end{eqnarray*}
+where $k = GmM = 10^{-3}M$, and $r = \sqrt{(q - q_*)^T(q - q_*)}$ is the distance between the planet and the star.
+
+We now introduce a measurement model, according to which our measurements suffer a normal error:
+$$
+  q_\mathrm{obs} \sim \mathrm{Normal}(q, \sigma I),
+$$
+with $\sigma$ the standard deviation and $I$ the $2 \times 2$  identity matrix.
+
+## Simulation
+
+The simulation can be done in Stan, using one of its numerical integrators.
+At $t = 0$, $q_0 = (1, 0)$ and $p_0 = (0, 1)$,
+and we set $k = 1$.
+```{r, fig.width=4, fig.height=3}
+mod <- cmdstan_model("model/planetary_motion_sim.stan")
+
+n <- 40
+sim <- mod$sample(data = list(n = n, sigma_x = 0.01, sigma_y = 0.01), 
+                  chains = 1, iter_warmup = 1, 
+                  iter_sampling = 2, seed = 123)
+
+r_sim <- read_stan_csv(sim$output_files())
+simulation <- rstan::extract(r_sim, pars = "q_obs")$q_obs[1, , ]
+
+q_x <- simulation[, 1]
+q_y <- simulation[, 2]
+
+q_obs <- array(NA, c(n, 2))
+q_obs[, 1] <- q_x
+q_obs[, 2] <- q_y
+
+sub_set <- seq(from = 1, to = 40, by = 2)
+plot <- ggplot(data = data.frame(q_x = q_x[sub_set], 
+                                 q_y = q_y[sub_set], 
+                                 time = sub_set), 
+               aes(x = q_x, y = q_y, label = time)) +  theme_bw() +
+  geom_text()
+plot
+```
+
+# Fitting a simple model and diagnosing inference
+
+A first attempt at fitting the complete model fails spectacularly: the chains do not converge and take a long time to run.
+This is an invitation to start with a simpler model.
+A useful simplification is more manageable but still exhibits the challenges we encounter with the complete model.
+The hope is that a fix in this simple context translates into a fix in a more sophisticated setting.
+This turns out to be quite true here where a simple one-parameter model allows us to understand the fundamentally multimodal nature of the model.
+
+There are many ways to simplify a model.
+A general approach is to fix some of the parameters, which we can easily do when working with simulated data.
+We fix all the parameters, except for $k$.
+We now want to characterize the posterior distribution,
+$$
+   p(k \mid q_\mathrm{obs}),
+$$
+by fitting the model with Stan.
+Since $k$ is a scalar, we could use quadrature but our goal is to understand the pathologies that frustrate our inference algorithm, so we stick to MCMC.
```

Comment:
Right. But doing a quadrature is an inference method, whereas plotting the distribution isn't (you could argue against this I suppose). The point is in 1-D, we don't need MCMC, because simpler inference methods exist.